Common/src/main/java/com/tabnineCommon/state/CompletionsState.kt

```diff
@@ -0,0 +1,31 @@
+package com.tabnineCommon.state
+
+import com.tabnineCommon.general.Utils
+import java.util.concurrent.Future
+import java.util.concurrent.TimeUnit
+
+object CompletionsState {
+    private var isCompletionsEnabled = true
+    private var timer: Future<*>? = null
+
+    fun setCompletionsEnabled(isEnabled: Boolean) {
+        isCompletionsEnabled = isEnabled
+        CompletionsStateNotifier.publish(isEnabled)
+
+        timer?.cancel(false)
+
+        if (!isCompletionsEnabled) {
+            timer = Utils.executeThread(
+                Runnable {
+                    setCompletionsEnabled(true)
+                },
+                1,
+                TimeUnit.HOURS
```

Comment:
is it not configurable at all currently? is it ok?

Reply:
yes, we snooze the completions for one hour